src/os/ui/filter/im/filterimporter.js

```diff
@@ -116,6 +116,11 @@ os.ui.filter.im.FilterImporter.prototype.processData = function() {
       // if we have a layer ID, we were passed some context from a filter window, so use it
       var impliedFilterable = os.ui.filter.getFilterableByType(this.layerId);
       var columns = impliedFilterable ? impliedFilterable.getFilterColumns() : null;
+      var layer = os.map.mapContainer.getLayer(this.layerId);
+
+      if (layer instanceof os.layer.Vector && columns == null) {
+        columns = os.im.action.getVectorLayerColumns(layer);
```

Comment:
I don't think we want this for generic filters, because filters are applied to server layers and should only match server-provided columns. This is what `getFilterColumns` will provide, where supported. We do not want to match layers that do not support server-side filtering, like file layers.

Filter actions operate on local columns, so in that specific case it makes sense to try getting columns from the layer. That specific case should be handled in `os.ui.im.action.FilterActionImporter`.